File: ueueo-framework/ueueo-domain-driven-desgin/src/main/java/com/ueueo/ddd/application/contracts/dtos/FullAuditedEntityDto.cs
using System;
using Volo.Abp.Auditing;

namespace Volo.Abp.Application.Dtos;

/**
 * This class can be inherited by DTO classes to implement <see cref="IFullAuditedObject"/> interface.
*/
[Serializable]
public abstract class FullAuditedEntityDto : AuditedEntityDto, IFullAuditedObject
{
     * <inheritdoc />
    public boolean IsDeleted;// { get; set; }

     * <inheritdoc />
    public ID DeleterId;// { get; set; }

     * <inheritdoc />
    public DateTime? DeletionTime;// { get; set; }
}

/**
 * This class can be inherited by DTO classes to implement <see cref="IFullAuditedObject"/> interface.
*/
 * <typeparam name="TPrimaryKey">Type of primary key</typeparam>
[Serializable]
public abstract class FullAuditedEntityDto<TPrimaryKey> : AuditedEntityDto<TPrimaryKey>, IFullAuditedObject
{
     * <inheritdoc />
    public boolean IsDeleted;// { get; set; }

     * <inheritdoc />
    public ID DeleterId;// { get; set; }

     * <inheritdoc />
    public DateTime? DeletionTime;// { get; set; }
}

File: abp-frameworks/ueueo-domain-driven-design/src/main/java/com/ueueo/ddd/application/contracts/dtos/FullAuditedEntityDto.java
package com.ueueo.ddd.application.contracts.dtos;

import com.ueueo.ID;
import com.ueueo.auditing.IFullAuditedObject;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.Date;

/**
 * This class can be inherited by DTO classes to implement <see cref="IFullAuditedObject"/> interface.
 */
@EqualsAndHashCode(callSuper = true)
@Data
public abstract class FullAuditedEntityDto extends AuditedEntityDto implements IFullAuditedObject {

    private boolean isDeleted;

    private ID deleterId;

    private Date deletionTime;
}

